model-optimizer/extensions/ops/Cast.py

```diff
@@ -55,15 +111,10 @@ def infer(node: Node):
         if node.has_and_set('stop_value_propagation'):
             return
         if node.in_node(0).has_valid('value'):
-            new_blob, finite_match_count, zero_match_count = convert_blob(node.in_node(0).value, dst_type)
-            node.out_port(0).data.set_value(new_blob)
-
-            if finite_match_count:
-                log.error(
-                    ("{} elements of {} were clipped to infinity while converting an input blob for node '{}' to {}. " +
-                     refer_to_faq_msg(76)).format(finite_match_count, new_blob.size, node.name, dst_type))
-            if zero_match_count:
-                log.warning(
-                    ("{} elements of {} were clipped to zero while converting an input blob for node '{}' to {}. " +
-                     refer_to_faq_msg(77)).format(zero_match_count, new_blob.size, node.name, dst_type))
+            value = node.in_node(0).value
 
+            if dst_type in [packed_U4, packed_I4]:  # custom types conversion
+                Cast.custom_type_casting_and_packing(node, value, dst_type)
+            else:
+                node.out_port(0).data.set_value(
+                    Cast.helper_value_propagation(node.soft_get('name', node.id), value, dst_type))
```

Comment:
Introduce variable `node_name` and use it in the `assert` in the beginning of this function as well

Reply:
Done